### PR TITLE
Add -F/--non-recursively option that controls whether to walk the directory recursively; add --json support for list_file

### PR DESCRIPTION
**Pull request by ryanluoo**
# Add -F/--non-recursively option
Now by default, `ios-deploy` will walk the directory recursively .  In our practice , the directory structure may be very complicated and the file operation may cost a lot of time .  

In most cases we  only want to access some specific folder , hope this option may be helpful 

# Add --json support for list_file
Now list file  option only output the  path of the a file.  
We add `--json` support to get all the meta data of a file , like create time, file size , etc. The meta data will be organized as dictionary . 
image

### CODE REVIEW

**Comment by gabebear** [comment]
I'm torn on whether the default should be recursive or non-recursive(e.g. adding a "--recursively" instead of "--non-recursively"). Is there any known benefit to doing it recursively? I don't know of any other people/projects using this code path, so I'm less worried about introducing regressions.

@shazron What do you think?

**Comment by ryanluoo** [comment]
@gabebear I think we should keep the original default behavior ,  in case someone is using the file browsing feature

**Comment by gabebear** [comment]
OK, keeping it the way you already have it is the safest option.

File: src/ios-deploy/ios-deploy.m [hunk 1]
```diff
@@ -1463,6 +1500,10 @@ AFCConnectionRef start_house_arrest_service(AMDeviceRef device) {
 
 void list_files_callback(AFCConnectionRef conn, const char *name, read_dir_cb_reason reason)
 {
+    if (_json_output) {
```
Comment (gabebear) [inline src/ios-deploy/ios-deploy.m]: We should probably gate the call to read_dir() rather than return early inside it's callback.

**Pull request merged**